Autobuild of docker image with CircleCI

thehajime opened the pull request:
This commit introduces autobuild triggered by pull requests and push to
master branch so that we can automate the test before uploading docker
hub.

The workflow to deploy docker images is:

- make changes to master branch, or merge a pull request to master
  branch
- (manually) tag to the commit on the master branch

This tag will trigger an autobuild on circleci and upload images to
docker hub.  Pull requests only trigger build without the deployment.

thehajime committed:
update freebsd-11 iso location

Signed-off-by: Hajime Tazaki <thehajime@gmail.com>

thehajime committed:
fix java installation issue

Signed-off-by: Hajime Tazaki <thehajime@gmail.com>

tavip reviewed (approved):
Awesome, thank you Hajime! 

Once we merge this, we should disable the docker hub automated builds, right?

thehajime committed:
use circleci instead of docker hub autobuild

This commit introduces autobuild triggered by pull requests and push to
master branch so that we can automate the test before uploading docker
hub.

The workflow to deploy docker images is:

- make changes to master branch, or merge a pull request to master
  branch
- (manually) tag to the commit on the master branch

This tag will trigger an autobuild on circleci and upload images to
docker hub.  Pull requests only trigger build without the deployment.

Signed-off-by: Hajime Tazaki <thehajime@gmail.com>

thehajime commented:
> Once we merge this, we should disable the docker hub automated builds, right?

yep.  I will work on that too.
